service/importer/service_file.go

```diff
@@ -4,70 +4,84 @@ import (
 	"fmt"
 	"io/ioutil"
 	"path/filepath"
+	"strings"
 
-	"github.com/mesg-foundation/core/service/importer/assets"
-	"github.com/xeipuuv/gojsonschema"
+	"github.com/go-playground/locales/en"
+	ut "github.com/go-playground/universal-translator"
+	"github.com/mesg-foundation/core/x/xvalidator"
+	validator "gopkg.in/go-playground/validator.v9"
+	en_translations "gopkg.in/go-playground/validator.v9/translations/en"
 	yaml "gopkg.in/yaml.v2"
 )
 
-// ConfigurationDependencyKey is the reserved key of the service's configuration in the dependencies array.
-const ConfigurationDependencyKey = "service"
+const (
+	namespacePrefix = "ServiceDefinition."
+)
 
 func readServiceFile(path string) ([]byte, error) {
 	file := filepath.Join(path, "mesg.yml")
 	return ioutil.ReadFile(file)
 }
 
 // validateServiceFile returns a list of warnings.
-func validateServiceFile(data []byte) ([]string, error) {
-	var body interface{}
-	if err := yaml.Unmarshal(data, &body); err != nil {
-		return nil, fmt.Errorf("parse mesg.yml error: %s", err)
-	}
-	body = convert(body)
-	result, err := validateServiceFileSchema(body, "service/importer/assets/schema.json")
-	if err != nil {
-		return nil, err
-	}
-	var warnings []string
-	if !result.Valid() {
-		for _, warning := range result.Errors() {
-			warnings = append(warnings, warning.String())
+func validateServiceFile(data []byte) []string {
+	var service ServiceDefinition
+	if err := yaml.UnmarshalStrict(data, &service); err != nil {
+		errs, ok := err.(*yaml.TypeError)
+		if !ok {
+			return []string{err.Error()}
 		}
+		return errs.Errors
 	}
-	if depKeyWarning := validateServiceFileDependencyKey(body); depKeyWarning != "" {
-		warnings = append(warnings, depKeyWarning)
-	}
-	return warnings, nil
+	return validateServiceStruct(&service)
 }
 
-func validateServiceFileDependencyKey(data interface{}) (warning string) {
-	s, _ := data.(map[string]interface{})
-	dep, _ := s["dependencies"].(map[string]interface{})
-	if dep[ConfigurationDependencyKey] != nil {
-		return fmt.Sprintf("cannot use %q as dependency key", ConfigurationDependencyKey)
-	}
-	return ""
-}
+func validateServiceStruct(service *ServiceDefinition) []string {
```

Comment:
Validation is done only for struct fields, but it need some few more validations (Volumes and dependencies etc...)

You can take a look 
https://github.com/mesg-foundation/core/blob/service-refactor/service/definition.go#L63

But it could be added with refactoring PR

Reply:
Done